implementation/interfacing_with_emlib.tex

```diff
@@ -164,32 +164,226 @@ \subsection{Naming Conventions}
+The \gls{rtc} was an easy module to write bindings for, it is used in several examples for timing purposes, as well as in the {\prog{sensor-tracker}}. \\
+
+\prog{timer}   &
+The Timer was one of the first modules to be ported, it worked as a proof-of-concept for the design of the bindings, as described earlier in this section. \\
+
+    \hline
+  \end{tabular}
+
+  \caption{Peripheral bindings for {\emlib}.}
+  \label{tab:emlib_peripheral_bindings}
+\end{table}
+
+
+\begin{table}[H]
+  \centering
+  \begin{tabular}{r|p{7cm}|p{2.1cm}}
+    \textbf{Example} & \textbf{Purpose} & \textbf{Uses} \\
+    \hline
+\prog{buttons\_int}  &
+Demonstrates interrupts by lighting a led on the {\chip{STK}} when the respective button has been pressed. &
+\gls{gpio}, IRQ \\
+
+\prog{rtc\_blink}    &
+Toggle a led with an interval of 2 seconds. &
+\gls{rtc}, \gls{gpio} \\
+
+\prog{energy\_modes} &
+Demonstrates the four stages of sleep on the {\gecko}. &
+\gls{emu}, \gls{gpio}, \gls{cmu}, IRQ \\
+
+\prog{uart}          &
+Demonstrates serial communication over \gls{usart} by echoing back every byte it receives. &
+\gls{usart}, \gls{gpio}, \gls{cmu} \\
+
+\prog{leuart}        &
+Similar example as \prog{uart}, but the \gls{cpu} is turned off and the functionality is moved to an interrupt handler instead. &
+\gls{leuart}, \gls{cmu}, IRQ, \gls{gpio}, \gls{emu} \\
+
+\prog{i2c}           &
+Sends and receives a data-buffer between two devices that supports \gls{i2c}. &
+\gls{i2c}, \gls{gpio}, \gls{cmu}, IRQ \\
+
+\prog{joystick}      &
+Reads analog signals generated by a Joystick that is connected to the {\chip{STK}}. &
+\gls{adc}, \gls{cmu} \\
+
+{\prog{dma}}           &
+Transfer a data-buffer from one memory location to another. &
+\gls{dma} \\
+
+\prog{light\_sense}  &
+Uses the {\chip{STK}}'s light sensor to measure the light intensity, and signal when it goes dark by lighting a LED. &
+\gls{lesense}, \gls{acmp}, \gls{gpio}, \gls{rtc}\\
+
+\prog{boxes}         &
+Demonstrates dynamic memory allocation with \code{Box}'es, provided by the Rust \lib{alloc} library. &
+\lib{alloc} \\
+
+\prog{vec}           &
+Demonstrates dynamically allocated strings and vectors from the \lib{collections} library. &
+\lib{collections} \\
+
+    \hline
+  \end{tabular}
+
+  \caption{Examples that demonstrates how the bindings work.}
+  \label{tab:emlib_examples}
+\end{table}
+
+\subsubsection{Emdrv}
+\label{sub:emdrv_bindings}
+
+As with {\emlib}, it was not a goal to fully support all available peripheral drivers that are available in \prog{emdrv}.
+The drivers that we have partially implemented bindings for are summarized in \autoref{tab:emdrv_bindings}, they are not that hard to grasp or use since they are all fairly small.
+The project includes two examples that demonstrate how to use the flash driver, they are summarized in \autoref{tab:emdrv_examples}  \todo{table needs to refer to section about IRQ. Evt. flytt noe av innholdet fra IRQ-seksjonen inn i denne seksjonen.}.
```

Comment:
what project? I think we should start this section by a complete overview of the RustyGecko libraries.